e2e-test/src/test/java/io/cdap/plugin/http/stepsdesign/HttpPlugin.java

```diff
@@ -0,0 +1,396 @@
+  public void toggleOAuthEnabledToTrue() {
+    HttpPluginActions.toggleOAuthEnabled();
+  }
+
+  @Then("Enter OAuth2 properties with blank {string}")
+  public void enterOAuthPropertiesWithBlank(String property) {
+    if (property.equalsIgnoreCase("authUrl")) {
+      HttpPluginActions.enterTokenUrl(E2ETestUtils.pluginProp("httpTokenUrl"));
+      HttpPluginActions.enterClientId(E2ETestUtils.pluginProp("httpClientId"));
+      HttpPluginActions.enterClientSecret(E2ETestUtils.pluginProp("httpClientSecret"));
+      HttpPluginActions.enterRefreshToken(E2ETestUtils.pluginProp("httpRefreshToken"));
+    } else if (property.equalsIgnoreCase("tokenUrl")) {
+      HttpPluginActions.enterAuthUrl(E2ETestUtils.pluginProp("httpAuthUrl"));
+      HttpPluginActions.enterClientId(E2ETestUtils.pluginProp("httpClientId"));
+      HttpPluginActions.enterClientSecret(E2ETestUtils.pluginProp("httpClientSecret"));
+      HttpPluginActions.enterRefreshToken(E2ETestUtils.pluginProp("httpRefreshToken"));
+    } else if (property.equalsIgnoreCase("clientId")) {
+      HttpPluginActions.enterAuthUrl(E2ETestUtils.pluginProp("httpAuthUrl"));
+      HttpPluginActions.enterTokenUrl(E2ETestUtils.pluginProp("httpTokenUrl"));
+      HttpPluginActions.enterClientSecret(E2ETestUtils.pluginProp("httpClientSecret"));
+      HttpPluginActions.enterRefreshToken(E2ETestUtils.pluginProp("httpRefreshToken"));
+    } else if (property.equalsIgnoreCase("clientSecret")) {
+      HttpPluginActions.enterAuthUrl(E2ETestUtils.pluginProp("httpAuthUrl"));
+      HttpPluginActions.enterTokenUrl(E2ETestUtils.pluginProp("httpTokenUrl"));
+      HttpPluginActions.enterClientId(E2ETestUtils.pluginProp("httpClientId"));
+      HttpPluginActions.enterRefreshToken(E2ETestUtils.pluginProp("httpRefreshToken"));
+    } else if (property.equalsIgnoreCase("refreshToken")) {
+      HttpPluginActions.enterAuthUrl(E2ETestUtils.pluginProp("httpAuthUrl"));
+      HttpPluginActions.enterTokenUrl(E2ETestUtils.pluginProp("httpTokenUrl"));
+      HttpPluginActions.enterClientId(E2ETestUtils.pluginProp("httpClientId"));
+      HttpPluginActions.enterClientSecret(E2ETestUtils.pluginProp("httpClientSecret"));
+    } else {
+      Assert.fail("Invalid Http OAuth2 property");
+    }
+  }
+
+  @Then("Validate mandatory property error for {string}")
+  public void validateMandatoryPropertyErrorFor(String property) {
+    HttpPluginActions.clickValidateButton();
+    SeleniumHelper.waitElementIsVisible(HttpPluginLocators.validateComplete, 5L);
+    E2ETestUtils.validateMandatoryPropertyError(property);
+  }
+
+  @Then("Validate mandatory property error for OAuth2 {string}")
+  public void validateMandatoryPropertyErrorForOAuth(String property) {
+    HttpPluginActions.clickValidateButton();
+    SeleniumHelper.waitElementIsVisible(HttpPluginLocators.validateComplete, 5L);
+    HttpPluginActions.validateOAuthPropertyError(property);
+  }
+
+  @Then("Validate HTTP properties")
+  public void validateHTTPProperties() {
```

Comment:
are these properties for the source or sink plugin? similar comment for all other http plugin configuration.

Reply:
This can be used for both sink and source as locators are same. For source/sink specific configurations we will prefix/suffix it with source/sink